modules/h3/bigquery/sql/H3_POLYFILL.sql

```diff
@@ -14,40 +14,71 @@ AS """
+    if (resolution < 0 || resolution > 16) {
         return null;
     }
 
+    const bboxA = [-180, -90, 0, 90]
+    const bboxB = [0, -90, 180, 90]
     const featureGeometry = JSON.parse(geojson)
-    let polygonCoordinates = [];
+    let polygonCoordinatesA = [];
+    let polygonCoordinatesB = [];
     switch(featureGeometry.type) {
         case 'GeometryCollection':
             featureGeometry.geometries.forEach(function (geom) {
                 if (geom.type === 'MultiPolygon') {
-                    polygonCoordinates = polygonCoordinates.concat(geom.coordinates);
+                    var clippedGeometryA = h3Lib.bboxClip(geom, bboxA).geometry;
+                    polygonCoordinatesA = polygonCoordinatesA.concat(clippedGeometryA.coordinates);
+                    var clippedGeometryB = h3Lib.bboxClip(geom, bboxB).geometry;
+                    polygonCoordinatesB = polygonCoordinatesB.concat(clippedGeometryB.coordinates);
                 } else if (geom.type === 'Polygon') {
-                    polygonCoordinates = polygonCoordinates.concat([geom.coordinates]);
+                    var clippedGeometryA = h3Lib.bboxClip(geom, bboxA).geometry;
+                    polygonCoordinatesA = polygonCoordinatesA.concat([clippedGeometryA.coordinates]);
+                    var clippedGeometryB = h3Lib.bboxClip(geom, bboxB).geometry;
+                    polygonCoordinatesB = polygonCoordinatesB.concat([clippedGeometryB.coordinates]);
+                }
+                else{
+                    var bboxGeom = h3Lib.bboxPolygon(h3Lib.bbox(geom)).geometry;
+                    var clippedGeometryA = h3Lib.bboxClip(bboxGeom, bboxA).geometry;
+                    polygonCoordinatesA = polygonCoordinatesA.concat([clippedGeometryA.coordinates]);
+                    var clippedGeometryB = h3Lib.bboxClip(bboxGeom, bboxB).geometry;
+                    polygonCoordinatesB = polygonCoordinatesB.concat([clippedGeometryB.coordinates]);
                 }
             });
         break;
         case 'MultiPolygon':
-            polygonCoordinates = featureGeometry.coordinates;
+            var clippedGeometryA = h3Lib.bboxClip(featureGeometry, bboxA).geometry;
+            polygonCoordinatesA = clippedGeometryA.coordinates;
+            var clippedGeometryB = h3Lib.bboxClip(featureGeometry, bboxB).geometry;
+            polygonCoordinatesB = clippedGeometryB.coordinates;
         break;
         case 'Polygon':
-            polygonCoordinates = [featureGeometry.coordinates];
+            var clippedGeometryA = h3Lib.bboxClip(featureGeometry, bboxA).geometry;
+            polygonCoordinatesA = [clippedGeometryA.coordinates];
+            var clippedGeometryB = h3Lib.bboxClip(featureGeometry, bboxB).geometry;
+            polygonCoordinatesB = [clippedGeometryB.coordinates];
         break;
         default:
-            return null;
+            var bboxGeom = h3Lib.bboxPolygon(h3Lib.bbox(featureGeometry)).geometry;
```

Comment:
Are we ok with the `LINESTRING` case? Is the BBOX used axis aligned? object-aligned?

Reply:
The idea here is that all geometries not previously supported are converted now to the their bbox, and then treated as a polygon. Is that what you meant?

Reply:
Not exactly. Let's see we have a horizontal line, both AABB and OABB are the same in those cases, but if the line is in a diagonal manner, the OABB is way greater, so the polyfill will be extended not only through the line but in the whole BBOX.

Reply:
Testing the h3-js polyfill function https://h3geo.org/docs/api/regions/#polyfill, I found it does not work with points or lines.

I have a workaround in mind, which is converting the points and lines into polygons, no bbox but something like a buffer, and then applying polyfill in the buffered section. Buffer distance depends on the resolution: https://h3geo.org/docs/core-library/restable/.

However, I would tackle that in another PR (will create a ticket for the backlog) and revert the bbox changes in this one to keep only the large polygons fix and the max resolution fix.